packages/Discretization/src/DTK_PointSearch_decl.hpp

```diff
@@ -0,0 +1,212 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#ifndef DTK_POINT_SEARCH_DECL_HPP
+#define DTK_POINT_SEARCH_DECL_HPP
+
+#include "DTK_ConfigDefs.hpp"
+#include <DTK_CellTypes.h>
+#include <DTK_DetailsBox.hpp>
+#include <DTK_DetailsPoint.hpp>
+#include <DTK_DistributedSearchTree.hpp>
+#include <DTK_PointInCell.hpp>
+
+#include <Kokkos_Core.hpp>
+#include <Teuchos_Comm.hpp>
+#include <Teuchos_RCP.hpp>
+#include <Tpetra_Distributor.hpp>
+
+#include <tuple>
+
+namespace DataTransferKit
+{
+template <typename DeviceType>
+class PointSearch
+{
+  public:
+    /**
+     * Constructor. The research of the points is done in the constructor but
+     * the results is not send back to the calling processor.
+     * @param comm
+     * @param cell_topologies_view
+     * @param cells vertices associated to each cell
+     * @param nodes_coordinates coordinates of all the nodes in the mesh
+     * @param points_coordinates coordinates in the physical frame of the points
+     * that we want to evaluate.
+     */
+    PointSearch(
+        Teuchos::RCP<const Teuchos::Comm<int>> comm,
+        Kokkos::View<DTK_CellTopology *, DeviceType> cell_topologies_view,
+        Kokkos::View<unsigned int *, DeviceType> cells,
+        Kokkos::View<double **, DeviceType> nodes_coordinates,
+        Kokkos::View<double **, DeviceType> points_coordinates );
```

Comment:
Please make it clear here that arguments 2, 3, and 4 are used to build the spatial index and argument 5 correspond to the spatial queries.  I am not sure whether "source" and "target" is appropriate here but prefixing the names might help understand what your constructor do.